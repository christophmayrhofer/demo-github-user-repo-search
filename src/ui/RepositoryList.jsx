import React, { PropTypes } from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import RepositoryItem from './RepositoryItem';

const RepositoryList = ({ repositories }) => (
  <div className="flex full-width">
    <If condition={repositories.count() === 0}>
      <div className="alert alert-info">
        This user does not have any public repositories.
      </div>
    </If>

    {repositories.map(repository => (
      <RepositoryItem
        key={repository.get('id')}
        name={repository.get('name')}
        description={repository.get('description')}
        language={repository.get('language')}
        htmlUrl={repository.get('html_url')}
      />
    ))}
  </div>
);

RepositoryList.propTypes = {
  repositories: ImmutablePropTypes.listOf(
    ImmutablePropTypes.contains({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      language: PropTypes.string.isRequired,
      html_url: PropTypes.string.isRequired,
    })
  ),
};

export default RepositoryList;
