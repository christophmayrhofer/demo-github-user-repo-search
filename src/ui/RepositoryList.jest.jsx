import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Immutable from 'immutable';
import RepositoryList from './RepositoryList';

describe('RepositoryList', () => {
  it('renders existing repositories', () => {
    const props = {
      repositories: Immutable.fromJS([
        {
          id: 1,
          name: 'RepositoryName 1',
          description: 'RepositoryDescription 1',
          language: 'RepositoryLanguage 1',
          html_url: 'RepositoryUrl 1',
        },
        {
          id: 2,
          name: 'RepositoryName 2',
          description: 'RepositoryDescription 2',
          language: 'RepositoryLanguage 2',
          html_url: 'RepositoryUrl 2',
        },
      ]),
    };

    const wrapper = shallow(
      <RepositoryList {...props} />
    );

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('renders message if the list contains no repositories', () => {
    const props = { repositories: Immutable.fromJS([]) };

    const wrapper = shallow(
      <RepositoryList {...props} />
    );

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
