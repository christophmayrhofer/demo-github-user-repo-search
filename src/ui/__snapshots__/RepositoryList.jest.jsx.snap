exports[`RepositoryList renders existing repositories 1`] = `
<div
  className="flex full-width">
  <RepositoryItem
    description="RepositoryDescription 1"
    htmlUrl="RepositoryUrl 1"
    language="RepositoryLanguage 1"
    name="RepositoryName 1" />
  <RepositoryItem
    description="RepositoryDescription 2"
    htmlUrl="RepositoryUrl 2"
    language="RepositoryLanguage 2"
    name="RepositoryName 2" />
</div>
`;

exports[`RepositoryList renders message if the list contains no repositories 1`] = `
<div
  className="flex full-width">
  <div
    className="alert alert-info">
    This user does not have any public repositories.
  </div>
</div>
`;
